cfme/infrastructure/pxe.py

```diff
@@ -1,96 +1,212 @@
+from widgetastic_manageiq import ManageIQTree, Input, ScriptBox, SummaryTable, Table
+from widgetastic_patternfly import Dropdown, Accordion, FlashMessages, BootstrapSelect, Button
+
+from cfme.base.login import BaseLoggedInPage
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
 from utils import conf
 from utils.datafile import load_data_file
-from utils.log import logger
 from utils.path import project_path
 from utils.update import Updateable
 from utils.wait import wait_for
 from utils.pretty import Pretty
 from utils.varmeth import variable
 
-cfg_btn = partial(tb.select, 'Configuration')
 
-pxe_server_table_exist = Table('//div[@id="records_div"]/table/tbody/tr/td')
-pxe_details_page = Region(locators=dict(
-    last_refreshed=InfoBlock("Basic Information", "Last Refreshed On"),
-    pxe_image_type=InfoBlock("Basic Information", "Type")
-))
+class PXEToolBar(View):
+    """
+    represents PXE toolbar and its controls
+    """
+    # todo: add back button later
+    configuration = Dropdown(text='Configuration')
+
+
+class PXESideBar(View):
+    """
+    represents left side bar. it usually contains navigation, filters, etc
+    """
+    @View.nested
+    class servers(Accordion):  # noqa
+        ACCORDION_NAME = "PXE Servers"
+        tree = ManageIQTree()
+
+    @View.nested
+    class templates(Accordion):  # noqa
+        ACCORDION_NAME = "Customization Templates"
+        tree = ManageIQTree()
+
+    @View.nested
+    class image_types(Accordion):  # noqa
+        ACCORDION_NAME = "System Image Types"
+        tree = ManageIQTree()
+
+    @View.nested
+    class datastores(Accordion):  # noqa
+        ACCORDION_NAME = "ISO Datastores"
+        tree = ManageIQTree()
+
+
+class PXEMainView(BaseLoggedInPage):
+    """
+    represents whole All PXE Servers page
+    """
+    flash = FlashMessages('.//div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+                          'contains(@class, "flash_text_div")]')
+    toolbar = View.nested(PXEToolBar)
+    sidebar = View.nested(PXESideBar)
+
+    @View.nested
+    class entities(View):  # noqa
+        """ will be filled in concrete classes"""
+
+    @property
+    def is_displayed(self):
+        return self.navigation.currently_selected == ['Compute', 'Infrastructure', 'PXE']
+
+
+class PXEServerEntity(ParametrizedView, ClickableMixin):
+    """
+     represents one PXE Server table row
+    """
+    PARAMETERS = ('name', )
+    ROOT = ParametrizedLocator('.//tr[./td[contains(normalize-space(.), {name|quote})]]')
+
+    checkbox = Checkbox(locator='./td/input[@type="checkbox"]')
+    name = Text(locator=ParametrizedString('./td[contains(normalize-space(.), {name|quote})]'))
+    last_refreshed = Text(locator=ParametrizedString('./td[contains(normalize-space(.), '
+                                                     '{name|quote})]/following-sibling::td'))
```

Comment:
Amount of entities is dynamically changed in this view. Besides that each entity has its own checkbox. As far as I remember Table widget cannot cope with such situation.